platform/kube/inject/configmap.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inject
+
+// NOTE: This tool only exists because kubernetes does not support
+// dynamic/out-of-tree admission controller for transparent proxy
+// injection. This file should be removed as soon as a proper kubernetes
+// admission controller is written for istio.
+
+import (
+	"fmt"
+
+	multierror "github.com/hashicorp/go-multierror"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	proxyconfig "istio.io/api/proxy/v1/config"
+	"istio.io/pilot/model"
+	"istio.io/pilot/proxy"
+)
+
+// GetMeshConfig fetches the ProxyMesh configuration from Kubernetes ConfigMap.
+func GetMeshConfig(kube kubernetes.Interface, namespace,
+	name string) (*v1.ConfigMap, *proxyconfig.ProxyMeshConfig, error) {
```

Comment:
it seems (or my search failed) that you always ignore the first argument, so why return it and change to _ everywhere ?

also next time consider doing a cp + commit + changes so git and the diff would know this is coming from 
platform/kube/inject/inject.go

also why the new file ?

Reply:
it's used here:
istioMeshConfigMap, istioMeshConfig, err := inject.GetMeshConfig(client, istioNamespace, meshConfigMapName)

Reply:
new file so it can be re-used in the initializer, Jason asked it to be like this.

Reply:
ok sorry I missed that / glossed over the "hack" section

afaik (unless there is some bazel magic) the file being in the same package there is 0 difference ?

Reply:
there is a new function, plus changes in the original fct. Please have a better look.

Reply:
would be easier to have a better look if it showed as a diff than a big green block